### PR TITLE
Fix vector clamp length methods to not accept negative values

### LINKED ISSUE #515: Vec2 clamp_length alters vector when length is within specified range
The `clamp_length` method in the `Vec2` incorrectly alters the vector components when the vector's length is within the specified `min` and `max` range. This behavior deviates from the expected functionality where the vector should remain unchanged if its length falls within the specified bounds.

```rust
let vec = Vec2::new(3., 4.);
let new_vec = vec.clamp_length(-10., 10.);
println!("{}", new_vec);
//the new_vec is [-6, -8]
```

While it is possible that this behavior is intentional in the current implementation, it seems counterintuitive for the method to alter the vector when its length is already within the specified range. Clarification on whether this is the expected behavior would be appreciated.

### PR DESCRIPTION
**Pull request by semihbkgr**
fixes #515

### CODE REVIEW

**Comment by bitshifter** [comment]
That looks good to me. Note that most code in glam is generated, see https://github.com/bitshifter/glam-rs/blob/main/codegen%2FREADME.md for how to run codegen.

Could you add a test for the problem you are fixing as well?

**Comment by bitshifter** [comment]
Sorry, I had some more thoughts on this which I've posted on the original issue.

**Comment by bitshifter** [comment]
GitHub is saying there are merge conflicts, not sure why.

You'd need to pull latest. Usually I rebase on top of latest and force push to the PR branch.

**Comment by bitshifter** [comment]
Nevermind, I think GitHub mobile was confused

**Comment by semihbkgr** [comment]
I got the same conflit on my local fork. To resolve, I just merged the latest `main` branch into this branch.

File: codegen/templates/vec.rs.tera [hunk 1]
```diff
@@ -1862,14 +1862,15 @@ impl {{ self_t }} {
...
+    /// Returns a vector with a length no less than `min` and no more than `max`.
     ///
     /// # Panics
     ///
-    /// Will panic if `min` is greater than `max` when `glam_assert` is enabled.
+    /// Will panic if `min` is greater than `max`, or if either `min` or `max` is negative, when `glam_assert` is enabled.
     #[inline]
     #[must_use]
     pub fn clamp_length(self, min: {{ scalar_t }}, max: {{ scalar_t }}) -> Self {
+        glam_assert!(0 as {{ scalar_t }} <= min);
```
Comment (bitshifter) [inline codegen/templates/vec.rs.tera:1873]: This method is only used in float vectors so it can be 0.0 without the need for a cast

**Review by bitshifter — CHANGES_REQUESTED** [review]

File: codegen/templates/vec.rs.tera (hunk 1 above)
Comment (semihbkgr) [reply codegen/templates/vec.rs.tera:1873]: ohh, sure 👍

**Review by bitshifter — APPROVED** [review]

**Comment by bitshifter** [comment]
Thanks for reporting this and the PR!

**Pull request merged**